File: backend-app/src/main/java/com/example/oauth/guest/repositories/GuestRepository.java
package com.example.oauth.guest.repositories;

import java.util.List;

import com.example.oauth.guest.models.Guest;
import com.example.oauth.user.models.UserEntity;
import org.springframework.data.repository.CrudRepository;

public interface GuestRepository extends CrudRepository<Guest, Integer> {
  public List<Guest> findAllByUser(UserEntity userEntity);
  public long countGuestsByUser(UserEntity userEntity);
  public long countAllBy();
  public long countGuestsByConfirmedEquals(String checkTrue);
}

File: backend-app/src/main/kotlin/com.example.oauth/guest/repositories/GuestRepository.kt
package com.example.oauth.guest.repositories

import com.example.oauth.guest.models.Guest
import com.example.oauth.user.models.UserEntity
import org.springframework.data.repository.CrudRepository

interface GuestRepository: CrudRepository<Guest, Int> {
    fun findAllByUser(userEntity: UserEntity?): List<Guest?>?
    fun countGuestsByUser(userEntity: UserEntity?): Long
    fun countAllBy(): Long
    fun countGuestsByConfirmedEquals(checkTrue: String?): Long
}